test/db2.connection.test.js

```diff
@@ -50,3 +50,47 @@ function generateDSN(config) {
     ';CurrentSchema=' + config.schema;
   return dsn;
 }
+
+describe('lazyConnect', function() {
+  it('should skip connect phase (lazyConnect = true)', function(done) {
+    var dsConfig = {
+      host: 'invalid-hostname',
+      port: 80,
+      database: 'invalid-database',
+      username: 'invalid-username',
+      password: 'invalid-password',
+      lazyConnect: true,
+    };
+    var ds = getDS(dsConfig);
+
+    var errTimeout = setTimeout(function() {
+      done();
+    }, 2000);
+    ds.on('error', function(err) {
+      clearTimeout(errTimeout);
```

Comment:
@qpresley I think this is not the test case that fails. (from the error log the second one fails)

I set timeout for an emitted event here to make it exit after a certain time, because this test case expects `error` event never fired.

So if `error` not fired then it only executes `done()` in setTimeout without trigger `done(err)`,
if `error` is fired then `done(err)` will return before timeout.